### PR TITLE
Add SLE 15 SP4 to Build Validation

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

Add support for SLE 15 SP4 to build validation.

I have run sanity checks with this PR to at least avoid obvious breakages.

## Links

Sibling PRs:
* Sumaform: uyuni-project/sumaform#1089
* Susemanager-ci: SUSE/susemanager-ci#537
* Infrastructure: https://gitlab.suse.de/galaxy/infrastructure/-/merge_requests/526

Ports:
* 4.1: SUSE/spacewalk#17751 (for code synchronization purposes only)
* 4.2: SUSE/spacewalk#17750

## Changelogs

- [x] No changelog needed

### CODE REVIEW

File: testsuite/features/build_validation/core/allcli_sanity.feature [hunk 1]
```diff
@@ -184,6 +184,27 @@ Feature: Sanity checks
     And "sle15sp3_ssh_minion" should communicate with the server using public interface
     And the clock from "sle15sp3_ssh_minion" should be exact
 
+@sle15sp4_client
```
Comment (srbarrios) [inline testsuite/features/build_validation/core/allcli_sanity.feature:187]: We should definitely convert this feature in a template, as we do for smoke tests.
So we don't need to care about it, when we include new clients.

Comment (srbarrios) [reply testsuite/features/build_validation/core/allcli_sanity.feature:187]: But ok for this PR.

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature [hunk 2]
```diff
@@ -116,6 +116,24 @@ Feature: Synchronize products in the products page of the Setup Wizard
     And I wait until I see "Selected channels/products were scheduled successfully for syncing." text
     And I wait until I see "SUSE Linux Enterprise Server 15 SP3 x86_64" product has been added
 
+@sle15sp4_minion
+  Scenario: Add SUSE Linux Enterprise Server 15 SP4
+    When I follow the left menu "Admin > Setup Wizard > Products"
+    And I wait until I do not see "Loading" text
+    And I enter "SUSE Linux Enterprise Server 15 SP4 x86_64" as the filtered product description
+    And I select "SUSE Linux Enterprise Server 15 SP4 x86_64" as a product
```
Comment (ktsamis) [inline testsuite/features/build_validation/reposync/srv_sync_all_products.feature:124]: This has BETA in the name, check for the exact string.

**Review by ktsamis — CHANGES_REQUESTED** [review]
Small change but overall looks great

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/build_validation/core/allcli_sanity.feature (hunk 1 above)
Comment (Bischoff) [reply testsuite/features/build_validation/core/allcli_sanity.feature:187]: why not - but not here

File: testsuite/features/build_validation/create_bootstrap_repositories/srv_create_bootstrap_repositories.feature [hunk 3]
```diff
@@ -98,6 +98,18 @@ Feature: Create bootstrap repositories
   Scenario: Create the bootstrap repository for a SLES 15 SP3 Salt SSH minion
     When I create the bootstrap repository for "sle15sp3_ssh_minion" on the server
 
+@sle15sp4_client
```
[the remark this answers is not in the document]
Comment (Bischoff) [reply testsuite/features/build_validation/create_bootstrap_repositories/srv_create_bootstrap_repositories.feature:101]: same answer

File: testsuite/features/support/constants.rb [hunk 4]
```diff
@@ -333,6 +344,21 @@
...
+    sle-manager-tools15-beta-pool-x86_64-sp4
+    sle-module-containers15-sp4-pool-x86_64
+    sle-module-basesystem15-sp4-pool-x86_64
+    sle-module-server-applications15-sp4-pool-x86_64
+    sle-product-sles15-sp4-updates-x86_64
+    sle-manager-tools15-updates-x86_64-sp4
+    sle-manager-tools15-beta-updates-x86_64-sp4
+    sle-module-containers15-sp4-updates-x86_64
+    sle-module-basesystem15-sp4-updates-x86_64
+    sle-module-server-applications15-sp4-updates-x86_64
```
Comment (srbarrios) [inline testsuite/features/support/constants.rb:361]: just be sure these are the correct channels

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature (hunk 2 above)
Comment (Bischoff) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature:124]: thanks for the catch!

File: testsuite/features/support/constants.rb (hunk 4 above)
Comment (Bischoff) [reply testsuite/features/support/constants.rb:361]: tried with your json, the list seems to match

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature [hunk 5]
```diff
@@ -116,6 +116,24 @@ Feature: Synchronize products in the products page of the Setup Wizard
...
     And I wait until I see "SUSE Linux Enterprise Server 15 SP3 x86_64" product has been added
 
+@sle15sp4_minion
+  Scenario: Add SUSE Linux Enterprise Server 15 SP4
+    When I follow the left menu "Admin > Setup Wizard > Products"
+    And I wait until I do not see "Loading" text
+    And I enter "SUSE Linux Enterprise Server 15 SP4 x86_64 (BETA)" as the filtered product description
+    And I select "SUSE Linux Enterprise Server 15 SP4 x86_64 (BETA)" as a product
+    Then I should see the "SUSE Linux Enterprise Server 15 SP4 x86_64 (BETA)" selected
+    When I open the sub-list of the product "SUSE Linux Enterprise Server 15 SP4 x86_64"
```
Comment (ktsamis) [inline testsuite/features/build_validation/reposync/srv_sync_all_products.feature:126]: are you sure these are not also for BETA?

Comment (ktsamis) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature:126]: wait yes this line needs BETA as well to open the list and then the submenus are fine as they are

Comment (Bischoff) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature:126]: ah one more

Comment (Bischoff) [reply testsuite/features/build_validation/reposync/srv_sync_all_products.feature]: done, thanks again

**Review by ktsamis — APPROVED** [review]

**Pull request merged**